lib/chef/provider/package/aur.rb

```diff
@@ -0,0 +1,104 @@
+#
+# Author:: Ryan Chipman (<rchipman@mit.edu>)
+# Copyright:: Copyright (c) 2015 Ryan Chipman
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider/package'
+require 'chef/mixin/command'
+require 'chef/resource/package'
+
+require 'net/http'
+require 'uri'
+require 'json'
+
+class Chef
+  class Provider
+    class Package
+      class AUR < Chef::Provider::Package
+
+        provides :aur_package, os: "linux"
+
+        def load_current_resource
+          @current_resource = Chef::Resource::Package.new(@new_resource.name)
+          @current_resource.package_name(@new_resource.package_name)
+
+          @current_resource.version(nil)
+
+          Chef::Log.debug("#{@new_resource} checking pacman for #{@new_resource.package_name}")
+          status = popen4("pacman -Qi #{@new_resource.package_name}") do |pid, stdin, stdout, stderr|
```

Comment:
Yeah, I could rewrite it with shell_out if that would be preferable. I used popen because that's what the pacman provider uses. Can you point me to docs on shell_out? I'm having a hard time finding them.